tensorpipe/CMakeLists.txt

```diff
@@ -5,6 +5,7 @@
 # LICENSE file in the root directory of this source tree.
 
 add_library(tensorpipe
+  ${TP_STATIC_OR_SHARED}
```

Comment:
From the CMake docs:
> If no type is given explicitly the type is STATIC or SHARED based on whether the current value of the variable BUILD_SHARED_LIBS is ON.

Maybe I am missing something, but this change does not seem to add anything?

Reply:
Indeed, *by default* this keeps the exact same behavior as before. But it allows downstream users (PyTorch), who may want to build their library as a dynamic library, to change how TensorPipe is built without changing that "global" flag.